File: TradebullsAngular/src/app/Models/Sales/StockModel.ts
export class Stock_FilterModel {
    BrandID:number = 0;
    ProductID:number = 0;
    ParentProductID?:number = 0;
    Sku:string = "";
    ProductName:string = "";
    ProductIDs:string = "";
    TransID: number = 0;
    TransType: string = "";
    Remarks: string = "";
    IsRemarks: number = -1;
}

export class Stock_ViewModel {
    StockItemID:number = 0;
    BrandID:number = 0;
    BrandName:string = "";
    ProductID:number = 0;
    ParentProductID?:number = 0;
    SKU:string = "";
    ProductName:string = "";
    Remarks:string = "";
    Qty:number = 0;
    UsedQty: number = 0;
}

export class Stock_View_response_Model {
    success:boolean = false;
    message:string = "";
    data:Stock_ViewModel[] | null = null;
}

export class Stock_Used_ViewModel {
    StockItemID:number = 0;
    BrandID:number = 0;
    BrandName:string = "";
    ProductID:number = 0;
    ParentProductID?:number = 0;
    SKU:string = "";
    ProductName:string = "";
    Remarks:string = "";
    Qty:number = 0;
}

export class Stock_Register_FilterModel {
    BrandID:string = "";
    ProductID:string = "";
    VariationID:string = "";
}

File: HandloomAPI/Models/Sales/StockModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace HandloomAPI.Models.Sales
{
    public class Stock_ViewModel
    {
        public long StockItemID { get; set; }
        public long BrandID { get; set; }
        public string BrandName { get; set; }
        public long ProductID { get; set; }
        public long ParentProductID { get; set; }
        public string SKU { get; set; }
        public string ProductName { get; set; }
        public string Remarks { get; set; }
        public decimal InQty { get; set; }
        public decimal OutQty { get; set; }
        public decimal Qty { get; set; }
    }

    public class Stock_Filter
    {
        public long BrandID { get; set; }
        public long ProductID { get; set; }
        public long ParentProductID { get; set; }
        public string Sku { get; set; }
        public string ProductName { get; set; }
        public string ProductIDs { get; set; }
        public long TransID { get; set; }
        public string TransType { get; set; }
        public string Remarks { get; set; }
        public int IsRemarks { get; set; }
    }

    public class Stock_Used_ViewModel {
        public long StockItemID { get; set; }
        public long BrandID { get; set; }
        public string BrandName { get; set; }
        public long ProductID { get; set; }
        public long ParentProductID { get; set; }
        public string SKU { get; set; }
        public string ProductName { get; set; }
        public string Remarks { get; set; }
        public decimal Qty { get; set; }
    }


    public class Stock_Register_Filter
    {
        public string BrandID { get; set; }
        public string ProductID { get; set; }
        public string VariationID { get; set; }
    }
    public class Stock_Register_ViewModel
    {
        public long StockItemID { get; set; }
        public long BrandID { get; set; }
        public string BrandName { get; set; }
        public long ProductID { get; set; }
        public long ParentProductID { get; set; }
        public string SKU { get; set; }
        public string ProductName { get; set; }
        public decimal InQty { get; set; }
        public decimal OutQty { get; set; }
        public decimal Qty { get; set; }
        public string Remarks { get; set; }
        public string ImageURL { get; set; }
    }
    public class Stock_Register_Item_ViewModel
    {
        public long StockItemInOutID { get; set; }
        public long TransID { get; set; }
        public string TransType { get; set; }
        public string TransNo { get; set; }
        public DateTime? TransDate { get; set; }
        public long TransChildID { get; set; }
        public long StockItemID { get; set; }
        public long BrandID { get; set; }
        public string BrandName { get; set; }
        public long ProductID { get; set; }
        public long ParentProductID { get; set; }
        public string SKU { get; set; }
        public string ProductName { get; set; }
        public string Remarks { get; set; }
        public decimal OutQty { get; set; }
        public decimal Qty { get; set; }
    }
}
